### PR TITLE
fix logic for when to add proxy to database url

### PR DESCRIPTION
**Pull request by pgwillia**
Add tests to catch regressions and rename the boolean test to be more 
meaningful and to miror the method name.

#1454

### CODE REVIEW

File: app/helpers/catalog_helper.rb [hunk 1]
```diff
@@ -37,7 +37,7 @@ def ual_electronic_access_url(location, url)
   end
 
   def database_electronic_access_url(document)
-    electronic_access_url(document['enableproxy_tesim'].first == '1', document['url_tesim'].first)
+    electronic_access_url(document['enableproxy_tesim'].first == '0', document['url_tesim'].first)
   end
 
   def electronic_access_url(test, url)
```
Comment (weiweishi) [inline app/helpers/catalog_helper.rb]: I wonder if we can change  "test" here to a more meaningful name of the variable? maybe `enable_proxy`?

Comment (pgwillia) [reply app/helpers/catalog_helper.rb]: I think I've improved this. What do you think?

**Comment by seanluyk** [comment]
@weiweishi we're starting to get helpdesk tickets and emails from users about this issue. Is it possible to expedite its review, and getting it into prod soon?

File: app/helpers/catalog_helper.rb (hunk 1 above)
Comment (weiweishi) [reply app/helpers/catalog_helper.rb]: This looks much better to me know

**Pull request merged**